### ISSUE TITLE
Style/RegexpLiteral complains about its EnforcedStyle instead of AllowInnerSlashes

### ISSUE THREAD

**Issue by milouse** [bbatsov/rubocop#3897]
If you write a regexp containing inner slashes, RuboCop will keeps complaining about using `%r{}` even if you asks for slashes as enforced style in your .rubocop.yml file.

--------

## Expected behavior

With `EnforcedStyle: slashes` or `EnforcedStyle: mixed` RuboCop should not display a warning urging you to use `%r{}` without more explanation. The message should be more explicit like: « you should use %r{} when your regexp contains inner slashes (preference AllowInnerSlashes) [Style/RegexpLitteral] »

## Actual behavior

RuboCop warns me with a weird « Use %r{} around regular expression [Style/RegexpLitteral] »

## Steps to reproduce the problem

Write the following lines in a new files : 

```ruby
tutu = 'To/to lapin'

tutu.gsub(/o\/t?/, 'a')

puts tutu
```

In the same folder, create a .rubocop.yml file containing:

```yaml
Style/RegexpLiteral:
  EnforcedStyle: slashes
```

## RuboCop version

```
$ rubocop -V
0.46.0 (using Parser 2.3.2.0, running on ruby 2.1.2 x86_64-linux)
```

**Comment by bbatsov** [comment]
Agreed. This should be improved.

**Comment by milouse** [comment]
If we agree on the message to be displayed, I can try to make a PR to help you :)

**Comment by bbatsov** [comment]
I'll have to review the behavior of the cop first. Honestly I don't remember its different behaviors.

**Comment by jonas054** [comment]
@milouse Actually, you need this configuration:
```yaml
Style/RegexpLiteral:
  EnforcedStyle: slashes
  AllowInnerSlashes: true
```
because with the default `AllowInnerSlashes: false`, the meaning of `EnforcedStyle: slashes` is that slashes are only enforced as long as they don't make it necessary to escape inner slashes.

See the last example in the relevant [section of the manual](http://rubocop.readthedocs.io/en/latest/cops_style/#styleregexpliteral).
